src/nspawn/nspawn-mount.c

```diff
@@ -597,7 +597,7 @@ int mount_all(const char *dest,
                         continue;
 
                 r = mkdir_userns_p(dest, where, 0755, mount_settings, uid_shift);
-                if (r < 0 && r != -EEXIST) {
+                if (r < 0 && r != -EEXIST && r != -EROFS) {
```

Comment:
hmm, so if you check r against to specific values please use !IN_SET(r, -EEXIST, -EROFS). 

that said, I think EROFS (unlike EEXIST) should result in a log message at LOG_DEBUG level. Even though it's safe to ignore it is useful to know that mkdir failed due to EROFS when trying to track down what goes on...

hence, I think this should be written as:

```c
…
if (r < 0 && r != -EEXIST) {
        log_errno_debug(fatal && r != -EROFS ? LOG_ERR : LOG_DEBUG, r, "Failed to create directory %s: %m", where);
        if (r != -EROFS) { /* if the root directory is read-only, and we couldn't create the directory hence then that's OK, let's continue, as mount() will then log as needed */
                if (fatal)
                        return r;
                continue;
        }
}
…
```